modules/@apostrophecms/piece-type/ui/apos/components/AposRelationshipFieldsManager.vue

```diff
@@ -0,0 +1,90 @@
+<template>
+  <AposModal
+    class="apos-doc-editor" :modal="modal"
+    :modal-title="modalTitle"
+    @inactive="modal.active = false" @show-modal="modal.showModal = true"
+    @esc="cancel" @no-modal="$emit('safe-close')"
+  >
+    <template #secondaryControls>
+      <AposButton
+        type="default" label="Exit"
+        @click="cancel"
+      />
+    </template>
+    <template #primaryControls>
+      <AposButton
+        type="primary" label="Finished"
+        :disabled="doc.hasErrors"
+        @click="submit"
+      />
+    </template>
+    <template #main>
+      <AposModalBody>
+        <template #bodyMain>
+          <AposModalTabsBody>
+            <div class="apos-doc-editor__body">
+              <AposSchema
+                v-if="docReady"
+                :schema="schema"
+                v-model="doc"
+              />
+            </div>
+          </AposModalTabsBody>
+        </template>
+      </AposModalBody>
+    </template>
+  </AposModal>
+</template>
+
+<script>
+import AposModalParentMixin from 'Modules/@apostrophecms/modal/mixins/AposModalParentMixin';
+
+export default {
+  name: 'AposRelationshipFieldsManager',
+  mixins: [
+    AposModalParentMixin
+  ],
+  props: {
+    schema: {
+      type: Array,
+      default() {
+        return [];
+      }
+    },
+    item: {
+      type: Object,
+      default() {
+        return {};
+      }
+    }
+  },
+  emits: [ 'safe-close' ],
+  data() {
+    return {
+      doc: {
+        data: {},
+        hasErrors: false
+      },
+      docReady: false,
+      modal: {
+        active: false,
+        type: 'overlay',
+        showModal: true
+      },
+      modalTitle: `Edit Relationship for ${this.item.title}`
+    };
+  },
+  async mounted() {
+    this.modal.active = true;
+    this.docReady = true;
+    this.doc.data = this.item._fields || {};
+  },
+  methods: {
+    async submit() {
+      const relationshipItem = this.item;
```

Comment:
You are modifying a sub-property of a prop. That's an antipattern in Vue.

You should implement the v-model pattern instead. Accept the original version of the relationship as the `value` prop, and emit a new version with `input`. Then you can use `v-model` to pass `_fields` into this component.

Reply:
Done